tcms/rpc/tests/test_testcase.py

```diff
@@ -651,7 +651,7 @@ def test_add_component_from_same_product_is_allowed(self):
         result = self.rpc_client.TestCase.add_component(
             self.test_case.pk, self.good_component.name
         )
-        self.assertEqual(result["component"][0], self.good_component.pk)
+        self.assertEqual(result["id"], self.good_component.pk)
```

Comment:
Can you cherry-pick https://github.com/kiwitcms/Kiwi/commit/101f725b3db90c6e2d3b0e6396958002a952a8dd instead and then add the front-end fixes as a separate commit inside this PR.

Reply:
I see that https://github.com/kiwitcms/Kiwi/commit/101f725b3db90c6e2d3b0e6396958002a952a8dd does not belongs to any branch. I have used https://github.com/kiwitcms/Kiwi/commit/ce3fb53bc4761b958bff9c63d0818d4e5909545b instead.